File: src/main/java/no/nav/familie/ks/sak/app/rest/Behandling/grunnlag/søknad/RestAktørTilknytningUtland.java
package no.nav.familie.ks.sak.app.rest.Behandling.grunnlag.søknad;

import no.nav.familie.ks.sak.app.behandling.domene.typer.AktørId;
import no.nav.familie.ks.sak.app.grunnlag.søknad.TilknytningTilUtland;

public class RestAktørTilknytningUtland {
    public AktørId aktør;
    public TilknytningTilUtland.TilknytningTilUtlandVerdier tilknytningTilUtland;
    public String tilknytningTilUtlandForklaring;

    public RestAktørTilknytningUtland(AktørId aktør, TilknytningTilUtland.TilknytningTilUtlandVerdier tilknytningTilUtland, String tilknytningTilUtlandForklaring) {
        this.aktør = aktør;
        this.tilknytningTilUtland = tilknytningTilUtland;
        this.tilknytningTilUtlandForklaring = tilknytningTilUtlandForklaring;
    }
}

File: src/main/java/no/nav/familie/ks/sak/app/rest/behandling/grunnlag/søknad/RestAktørTilknytningUtland.kt
package no.nav.familie.ks.sak.app.rest.behandling.grunnlag.søknad

import no.nav.familie.ks.kontrakter.søknad.TilknytningTilUtlandVerdier
import no.nav.familie.ks.sak.app.behandling.domene.grunnlag.søknad.AktørTilknytningUtland

data class RestAktørTilknytningUtland(
    val fødselsnummer: String,
    val tilknytningTilUtland: TilknytningTilUtlandVerdier,
    val tilknytningTilUtlandForklaring: String)

fun AktørTilknytningUtland.toRestAktørTilknytningUtland() = RestAktørTilknytningUtland(
        fødselsnummer = this.fødselsnummer,
        tilknytningTilUtland = this.tilknytningTilUtland,
        tilknytningTilUtlandForklaring = this.tilknytningTilUtlandForklaring
)
